### PR TITLE
Issue/1874 Added preemption priority classes

### LINKED ISSUE #1874: Spike: Look into Kubernetes pod priority
### Problem description
As of a recently, Kubernetes pod priority has moved into beta:

https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/

We should look into how we can use this - probably something along the lines of making sure the gateway has the most priority, then elasticsearch/db, then public api pods, then minions/connectors.

Desired result:
Spike & do - figure out how to change the configuration and change it

### PR DESCRIPTION
**Pull request by nahidakbar**
### What this PR does

Fixes #1874 

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

File: deploy/helm/magda/charts/indexer/templates/deployment.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ spec:
       labels:
         service: indexer
     spec:
+      priorityClassName: magda-8
```
Comment (AlexGilleran) [inline deploy/helm/magda/charts/indexer/templates/deployment.yaml]: The indexer should probably sit a level below `api`s given that it's fine for it to be bumped at any time (it'll resume its hook and catch up when it restarts)

File: deploy/helm/magda/charts/session-db/templates/statefulset.yaml [hunk 2]
```diff
@@ -14,6 +14,7 @@ spec:
       labels:
         service: session-db
     spec:
+      priorityClassName: magda-10
```
Comment (AlexGilleran) [inline deploy/helm/magda/charts/session-db/templates/statefulset.yaml]: I don't think the session db needs to be higher than the other dbs - if it goes down most things will (should?) still work, it's just signed in functionality that'll fail from a user's point of view, and stuff within the system like crawling, which can wait.

File: deploy/helm/magda/charts/web-server/templates/deployment.yaml [hunk 3]
```diff
@@ -12,6 +12,7 @@ spec:
       labels:
         service: web
     spec:
+      priorityClassName: magda-7
```
Comment (AlexGilleran) [inline deploy/helm/magda/charts/web-server/templates/deployment.yaml]: I would think this is as important as the APIs?

File: deploy/helm/magda/charts/elasticsearch/templates/data-statefulset.yml [hunk 4]
```diff
@@ -18,6 +18,7 @@ spec:
         component: elasticsearch
         role: data
     spec:
+      priorityClassName: magda-9
```
Comment (AlexGilleran) [inline deploy/helm/magda/charts/elasticsearch/templates/data-statefulset.yml]: should also add a high priority to es masters 👍

**Comment by nahidakbar** [comment]
adjusted @AlexGilleran

**Pull request merged**